pages/categorization.vue

```diff
@@ -0,0 +1,521 @@
+				}
+				// Uses just tsv data
+				else {
+
+					console.log("Point 3a");
+
+					// 1. Produce an array of dicts
+					var tsvDictArray = [];
+
+					// A. Each dict has a header entry from the tsv file
+					let tsvIndex = 0;
+					for ( let headerField in tsvFile[0] ) {
+
+						tsvIndex += 1;
+
+						// I. Save a new dict for this column and description
+						tsvDictArray.push({
+							"column": headerField,
+							"description": "",
+							"primary-key": tsvIndex - 1
+						});
+					}
+
+					console.log("Point 3b");
+
+					// 2. Save the table in this component's data
+					this.$store.dispatch("saveTableData", tsvDictArray);
+				}
+
+				return this.tableData();
+			}
+		},
+
+		methods: {
+
+			annotationPageAccess(p_enable) {
+
+				console.log("Annotation page access with p_enable: " + p_enable);
+				
+				// 1. Enable/disable access to the annotation page on the nav bar
+				for ( let index = 0; index < this.navItemsState.length; index++ ) {
+					
+					// A. Look for the annotation nav item
+					if ( this.pageNames.annotation.pageName == this.navItemsState[index].pageInfo.pageName ) {
+
+						console.log("Found annotation pagename");
+						
+						// i. Enable/disable the annotation nav item
+						this.navItemsState[index].enabled = p_enable;
+						
+						break;
+					}
+				}
+
+				// 2. Enable/disable the next step button
+				this.readyForNextStepFlag = p_enable;
+
+				// 3. Change the next step button's color
+				this.nextPageButtonColor = ( p_enable ) ? "success" : "secondary";
+			},			
+
+			changeToNewState() {
+
+				// 1. Determine page state from data contents
+				let newState = this.determineState();
+
+				// 2. Change the page to the determined state
+				this.changeState(newState);
+			},
+
+			determineState() {
+
+				// 1. Reset the state to default
+				let newState = this.possibleStates.STATE_NOCATEGORIES_PAINTED;
+
+				// 2. Count the number of painted rows in the table
+				let paintedRowsCount = this.countPaintedColumns();
+
+				// 3. Add appropriate state flags based on data contents
+				if ( paintedRowsCount > 0 )
+					newState |= this.possibleStates.STATE_ATLEASTONE_CATEGORY_PAINTED;
+
+				return newState;
+			},
+
+			changeState(p_state) {
+
+				// 1. Trigger the behavior for the requested state change
+				switch ( p_state ) {
+
+					// A. Handle changes for when at least one category is painted
+					case this.possibleStates.STATE_ATLEASTONE_CATEGORY_PAINTED:
+						this.changeState_AtLeastOneCategoryPainted();
+						break;
+
+					// B. Handle changes for no categories are painted
+					default:
+						this.changeState_NoCategoriesPainted();
+						break;
+				}
+			},
+
+			changeState_AtLeastOneCategoryPainted() {
+
+				// Enable access to the annotation page
+				this.annotationPageAccess(true);
+			},
+
+			changeState_NoCategoriesPainted() {
+
+				// Disable access to the annotation page
+				this.annotationPageAccess(false);
+			},
+
+			countPaintedColumns() {
```

Comment:
I like this a lot! We could use this in a future iteration to display the number of columns painted by category next to the category selector table on the left. Could give users a quick way to see whether they got a plausible number of columns annotated.